test/usecases/nonlinear/test_nonlin.py

```diff
@@ -1,15 +1,25 @@
 from neuron import h, gui
 
 
-def test_nonlin():
+def check_solve(make_inst):
     s = h.Section()
     s.insert("nonlin")
-    inst = s(0.5).nonlin
+
+    inst = make_inst(s)
 
     x = inst.solve()
     error = inst.residual(x)
     assert abs(error) < 1e-9, f"{x = }, {error = }"
```

Comment:
Since you're modifying this file, maybe consider changing this call to `numpy.testing.assert_allclose` or similar.